lib/chat/bloc/provider/ChatProvider.dart

```diff
@@ -280,22 +271,62 @@ class ChatProvider implements ViewModelProvider<ChatViewModel> {
     );
   }
 
-  void _updateAvatarVisibility() {
+  void _updatePreviousMessages() {
```

Comment:
I think we can make this simpler with the concept of sections. but lets save that for a refactor

Reply:
Yes, agree. The section feature we have talked yesterday could be a good point.
Yes we will leave it for the refactor.